changelog/906.feature.rst

```diff
@@ -0,0 +1,4 @@
+Add application role connection metadata types.
```

Comment:
I don't think "types" should be there, what do you think?

Reply:
That doesn't really make sense, I agree :thinking:
updated the changelog in https://github.com/DisnakeDev/disnake/pull/906/commits/39680cb116ba629bbd62331b2a825216cbd5ebbc